bundles/com.zeligsoft.domain.dds4ccm/src/com/zeligsoft/domain/dds4ccm/utils/DDS4CCMMigrationModelTypeUtil.java

```diff
@@ -69,35 +92,268 @@ private static boolean migrateAll(Model model, boolean dryrun) {
+			return ConnectorType.CORBA4CCM_Connector.name();
+		}else{
+			return "";
+		}
+	}
+	
+	/**
+	 * Migrate property of a model element from ATCD to AXCIOMA  
+	 * 
+	 * @param e
+	 *         Element instance to be migrated 
+	 *          
+	 */	
+	private static void migrateToAxciomaProperty(Element e){
+		if(ZDLUtil.isZDLConcept(e, CCMNames.RESOURCE)){
+			Resource resource = ZDLFactoryRegistry.INSTANCE.create(e, Resource.class);
+			List<String> oldStrings = new ArrayList<String>(resource.getResourceType());
+			// ZDLUtil is used for setting value due to the limitation of the api code
+			ZDLUtil.setValue(e, CCMNames.REQUIREMENT_SATISFIER, CCMNames.REQUIREMENT_SATISFIER__RESOURCE_TYPE, null);				
+			
+			for(String s : oldStrings) {
+				resource.addResourceType(getAxciomaStr(s));
+			}
+		}else if(ZDLUtil.isZDLConcept(e, CCMNames.PROPERTY)){
+			com.zeligsoft.domain.omg.ccm.api.CCM_Target.Property oldProperty =  ZDLFactoryRegistry.INSTANCE.create(e, com.zeligsoft.domain.omg.ccm.api.CCM_Target.Property.class);
+			oldProperty.setName(getAxciomaStr(oldProperty.getName()));
+		}else if(ZDLUtil.isZDLConcept(e, CCMNames.SATISFIER_PROPERTY)){
+			SatisfierProperty oldSatisfierProperty =  ZDLFactoryRegistry.INSTANCE.create(e, SatisfierProperty.class);
+			ZDLUtil.setValue(e, CCMNames.SATISFIER_PROPERTY, CCMNames.SATISFIER_PROPERTY__VALUE, getAxciomaStr((String)ZDLUtil.getValue(e, CCMNames.SATISFIER_PROPERTY, CCMNames.SATISFIER_PROPERTY__VALUE)));
+			oldSatisfierProperty.setName(getAxciomaStr(oldSatisfierProperty.getName()));
+		}
+	}
+	
+	/**
+	 * Retrieves 'ConnectorDef' object for the default value of connectorType set in the preference page
+	 * 
+	 * @param defaultConnectorType
+	 *          Preferred value of 'ConnectorType' for a sync/async port for being used during migration 
+	 * @param model
+	 * 			an atcd model that is being migrated           
+	 * @return a 'ConnectorDef' object that corresponds to the default value of the 'ConnectorType' 
+	 *         
+	 */  
+	private static ConnectorDef getDefaultConnectorDef(String defaultConnectorType, Model model){
+		
+		String pathmapConnectorLib = getPathmapConnectorLib(defaultConnectorType);
+		URI uriConnectorLib = URI.createURI(pathmapConnectorLib);
+		Package packageConnectorLib = getImportedConnectorPackage(model, uriConnectorLib);
+		ConnectorDef connectorDef = getConnectorDef(packageConnectorLib);
+		
+		return connectorDef;
+	}
+	
+	/**
+	 * Retrieves pathmap of a connector library
+	 * 
+	 * @param preferredConnectorType
+	 *          Preferred value of 'ConnectorType' to be used during migration 
+	 *            
+	 * @return pathmap of a connector library which corresponds to the input connectorType value  
+	 *         
+	 */
+	
+	private static String getPathmapConnectorLib(String preferredConnectorType){
+		
+		String pathmapConnectorLib = "";
+		
+		String pathmapAMIConnector = "pathmap://DDS4CCM_AMI_LIBRARIES/CCM_AMI.uml";
```

Comment:
How many places in our code do we have knowledge that there are two model libraries for Axcioma connector types? If the answer is more than one, we should refactor so that there is one. We fully expect that, in the future, there will be other connector types added. If we have code sprinkled all over the place to handle them, then we run a very high risk that we will add future connector types incorrectly, or incompletely.